src/Gui/Document.cpp

```diff
@@ -1476,5 +1477,22 @@ void Document::handleChildren3D(ViewProvider* viewProvider)
                 }
             }
         }
+    } else if (viewProvider && viewProvider->isDerivedFrom(ViewProviderDocumentObjectGroup::getClassTypeId())) {
+
+        ViewProviderDocumentObject* vp = static_cast<ViewProviderDocumentObject *>(viewProvider);
+        auto* ext = vp->getObject()->getExtensionByType<App::GroupExtension>();

```

Comment:
I needed DocumentObject only to get children, but I've found the shorter way: ViewProvider->ClaimChildren().
Also I've added ViewProviderDocumentObjectGroup extension check inside my else if check.